src/backend/distributed/deparser/citus_ruleutils.c

```diff
@@ -739,6 +740,12 @@ deparse_shard_index_statement(IndexStmt *origStmt, Oid distrelid, int64 shardid,
 												relationName),
 					 indexStmt->accessMethod);
 
+	/* switch to search_path = pg_catalog before deparsing index columns */
+	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
```

Comment:
The other comment seems better but I would suggest creating a method such as `SwitchToPgCatalogSearchPath`:

```
	/*
	 * Set search_path to NIL so that all objects outside of pg_catalog will be
	 * schema-prefixed. pg_catalog will be added automatically when we call
	 * PushOverrideSearchPath(), since we set addCatalog to true;
	 */
```
So the function could be:


```
/*
 * SwitchToPgCatalogSearchPath switches to pg_catalog.
 * 
 */
void SwitchToPgCatalogSearchPath() {
	/*
	 * Set search_path to NIL so that all objects outside of pg_catalog will be
	 * schema-prefixed. pg_catalog will be added automatically when we call
	 * PushOverrideSearchPath(), since we set addCatalog to true;
	 */
	OverrideSearchPath *overridePath = GetOverrideSearchPath(CurrentMemoryContext);
	overridePath->schemas = NIL;
	overridePath->addCatalog = true;
	PushOverrideSearchPath(overridePath);
}
```

And we can use it both here and in `GetForeignConstraintCommandsToReferenceTable`

Reply:
looks like we already have such a function :)
PushOverrideEmptySearchPath